components/movementsensor/gpsnmea/nmeaParser.go

```diff
@@ -49,85 +49,141 @@ func (g *GPSData) ParseAndUpdate(line string) error {
 	if err != nil {
 		return multierr.Combine(errs, err)
 	}
-	// Most receivers support at least the following sentence types: GSV, RMC, GSA, GGA, GLL, VTG, GNS
-	if gsv, ok := s.(nmea.GSV); ok {
-		// GSV provides the number of satellites in view
-		g.SatsInView = int(gsv.NumberSVsInView)
-	} else if rmc, ok := s.(nmea.RMC); ok {
-		// RMC provides validity, lon/lat, and ground speed.
-		if rmc.Validity == "A" {
-			g.valid = true
-		} else if rmc.Validity == "V" {
-			g.valid = false
-			errs = multierr.Combine(errs, errInvalidFix(rmc.Type, rmc.Validity, "A"))
-		}
-		if g.valid {
-			g.Speed = rmc.Speed * knotsToMPerSec
-			g.Location = geo.NewPoint(rmc.Latitude, rmc.Longitude)
-		}
-	} else if gsa, ok := s.(nmea.GSA); ok {
-		// GSA gives horizontal and vertical accuracy, and also describes the type of lock- invalid, 2d, or 3d.
-		switch gsa.FixType {
-		case "1":
-			// No fix
+
+	errs = g.updateData(s)
+
+	if g.Location == nil {
+		g.Location = geo.NewPoint(math.NaN(), math.NaN())
+		errs = multierr.Combine(errs, errors.New("no Location parsed for nmea gps, using default value of lat: NaN, long: NaN"))
+		return errs
+	}
+
+	return nil
+}
+
+func (g *GPSData) updateData(s nmea.Sentence) error {
+	var errs error
+
+	switch sentence := s.(type) {
+	case nmea.GSV:
+		errs = g.updateGSV(sentence)
+	case nmea.RMC:
+		errs = g.updateRMC(sentence)
+	case nmea.GSA:
+		errs = g.updateGSA(sentence)
+	case nmea.GGA:
+		errs = g.updateGGA(sentence)
+	case nmea.GLL:
+		errs = g.updateGLL(sentence)
+	case nmea.VTG:
+		errs = g.updateVTG(sentence)
+	case nmea.GNS:
+		errs = g.updateGNS(sentence)
+	}
+
+	return errs
+}
+
+//nolint
+func (g *GPSData) updateGSV(gsv nmea.GSV) error {
+	// GSV provides the number of satellites in view
+	g.SatsInView = int(gsv.NumberSVsInView)
+	return nil
+}
+
+func (g *GPSData) updateRMC(rmc nmea.RMC) error {
```

Comment:
It would be helpful if there was a comment above each of these functions saying what each message type provides, similar to the one on GSV.